cryostatio/cryostat-web
# chore(eslint): add eslint plugins and rules to sort imports and remove unused imports

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #689

## Description of the change:
Adds an eslint plugin to check if any imports and unused as well as if there are any unused vars.

## Motivation for the change:
*This change is helpful because it is helpful to remove unused dependencies without going through every file.

## How to manually test:
1. Checkout pr
2. `yarn install`
3. Go to TargetSelect.test.tsx and add an import (finalize -> `import { of, finalize } from 'rxjs';`
4. run `yarn eslint`
5. Notice that the output flags the import as an error.
6. Run `yarn eslint:apply`
7. Notice the import is gone. 

## Other:
I haven't applied the eslint --fix yet which would also start to sort imports. I think it would touch a lot of files.
Also, I'm not sure why the "compiler" has started to complain about react 18 changes (implicit children), but I fixed them anyways.
